4-forms-validations/index.md

```diff
@@ -1 +1,115 @@
+
+![image](https://user-images.githubusercontent.com/36536646/79612762-e7408f00-80c2-11ea-9515-895551579124.png)
+
+## How to structure a web form
+
+* The `<form>` element formally defines a form and attributes that determine the form's behavior.
+* The URL that processes the form submission. This value can be overridden by a formaction attribute on a <button>, <input type="submit">, or <input type="image"> element.
+  
+```html
+    <form action="">
+        <!--form content-->
+    </form>
+```
+
+* It's strictly forbidden to nest a form inside another form
+* Don't forget *accessibility*.
+
+Some tips:
+
+* The `<label>` element is the formal way to define a label for an HTML form widget.
+* *Accessibility*: screen readers will speak a form element label along with any related instructions. 
+
+```html
+<label for="name">
+        Name: <input type="text" id="name" name="user_name">
+        <abbr title="required" aria-label="required">*</abbr>
+ </label>
+```
+
+* *type* attribute in the `<input>` tag [types](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input).
+* *required* attribute in the `<input>` tag.
+
+* Nest the `<input>` tag to the `<label>` tag.
+```html
+  <label for="name">
+    Name: <input type="text" id="name" name="user_name">
+  </label>
+```
+
+* The `<fieldset>` element is a convenient way to create groups of widgets that share the same purpose (styling and semantic). 
+* You can label a `<fieldset>` by including a `<legend>` element.
+* Select (5 or more options...)
+
+```html
+<fieldset>
+        <legend>Choose your city:</legend>
+        <select id="city">
+            <option value="Tunja">Tunja</option>
+            <option value="Bogota">Bogota</option>
+            <option value="Cali">Cali</option>
+            <option value="Bucaramanga">Bucaramanga</option>
+            <option value="Medellin">Medellin</option>
+            <option value="Pasto">Pasto</option>
+        </select>
+ </fieldset>
+```
+
+* Group radio buttons in fieldsets
+
+```html
+  <fieldset>
+        <legend>Choose your bootcamp:</legend>
+        <input id="basic" type="radio" name="bootcamp" value="basic">
+        <label for="basic">Basic web development</label>
+        <input id="advanced" type="radio" name="bootcamp" value="advanced">
+        <label for="advanced">Advanced web development</label>
+    </fieldset>
+```
+
+* Rule of thumb: If it's a long list or the alternatives aren't that important, use a select.
+
+## Validation
+
+There are two different types of client-side validation that you'll encounter on the web:
+
+* Built-in form validation: uses HTML5 form validation features. Built-in form validation has better performance than JavaScript, but it is not as customizable as JavaScript validation.
+
+```html
+    <label for="number">How old are you?</label>
+    <input type="number" id="number" name="amount" value="1" min="1" max="100">
+```
+
```

Comment:
Include examples of each as in https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation